src/workloads/execution/TimeSeriesRangeDelete.yml

```diff
@@ -0,0 +1,132 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/server-execution"
+Description: |
+  This test establishes a baseline for time-based range deletes on time-series collections.
+
+  We insert 1000 independent series with 100 buckets in each series, and each bucket has 100
+  documents.
+
+  The buckets have meta fields with values from 0 to 999, each of which has 100 measurements with
+  timestamps within an hour with 36-second intervals. For example, in a bucket, the measurements
+  look like:
+  {t: 2023-01-01T5:00:00, m: 42, ...},
+  {t: 2023-01-01T5:00:36, m: 42, ...},
+  ...
+  {t: 2023-01-01T5:59:24, m: 42, ...}
+
+  Then we delete data that spans across three buckets for each series, causing two partial-bucket
+  deletions and one full-bucket deletion. This tests the use case of data correction where the
+  application deletes data within some time ranges.
+
+GlobalDefaults:
+  dbname: &db test
+  coll: &coll Collection0
+  nop: &Nop {Nop: true}
+  MaxPhases: &MaxPhases 4
+
+Actors:
+- Name: CreateTimeSeriesCollection
+  Type: RunCommand
+  Threads: 1
+  Phases:
+  OnlyActiveInPhases:
+    Active: [0]
+    NopInPhasesUpTo: *MaxPhases
+    PhaseConfig:
+      Repeat: 1
+      Database: *db
+      Operation:
+        OperationName: RunCommand
+        OperationCommand:
+          {create: *coll, timeseries: {timeField: "t", metaField: "sensorId"}}
+
+- Name: InsertData
+  Type: Loader
+  Threads: 1
+  Phases:
+    OnlyActiveInPhases:
+      Active: [1]
+      NopInPhasesUpTo: *MaxPhases
+      PhaseConfig:
+        Repeat: 1
+        Database: *db
+        Collection: *coll
+        Threads: 1
+        CollectionCount: 1
+        DocumentCount: 10_000_000
+        BatchSize: 30000
+        Document:
+          t: {^Repeat: {count: 1000, fromGenerator: {^IncDate: {start: "2023-01-01", step: 36000}}}}
```

Comment:
hmmm, based of of my previous comment it looks like this count might have to be 10_000 ? or the docCount should be 1_000_000? But I might also be a tad confused 😅

Reply:
What this line does is it generates 1000 entries for each timestamp. So we insert timestamp1 for all 1000 devices first, and then timestamp2, and so on.

Reply:
I see. So we have something like this occurring: 

```
for docId in range(0, DocumentCount=10_000_000):
    for ts in range(0, *Initial ts + step x 1000*, step=36000):
        for meta in range(0, ofLength=1000):
            coll.insert({_id, docId, t: ts, "sensorId": meta,...})
```

Reply:
Added inline comments.